### DISCUSSION TITLE
HypreILU and HYPRE_ILUSetLevelOfFill

### DISCUSSION THREAD

**Discussion by rprat-pro** [thelfer/mfem-mgis#63] [Ideas]
I wanted to customize HypreILU for my example with: 

> HYPRE_Int HYPRE_ILUSetLevelOfFill([HYPRE_Solver](https://hypre.readthedocs.io/en/latest/api-sol-krylov.html#_CPPv412HYPRE_Solver)[](https://hypre.readthedocs.io/en/latest/api-sol-parcsr.html#_CPPv423HYPRE_ILUSetLevelOfFill12HYPRE_Solver9HYPRE_Int) solver, HYPRE_Int lfil)
> 
>     (Optional) Set the level of fill k, for level-based ILU(k) The default is 0 (for ILU(0)).

I tested it on my test case and this feature can reduce the solve runtime by 2 on 128 MPI processes over the version without this option (or be 4 times slower).

Maybe we could add this option in mfem-mgis.

Raphaël

**Comment by thelfer** [comment]
@rprat-pro No problem for that. Do you need help how to use the `Parameter` class to pass options to the preconditioner in the `LinearFactory` class ?

**Comment by rprat-pro** [comment]
Looks like my changes work correctly in my example. I did a merge request.
